src/app/services/nodeRewardsProgram.service.js

```diff
@@ -0,0 +1,132 @@
+import nem from 'nem-sdk';
+
+const NODE_REWARDS_PROGRAM_API_BASE_URL = 'https://supernodesapi.nem.io';
+
+/** Service to enroll in the Node Rewards Program and to get the status related information */
+class NodeRewardsProgram {
+
+    /**
+     * Initialize dependencies and properties
+     *
+     * @params {services} - Angular services to inject
+     */
+    constructor($localStorage, $filter, Wallet) {
+        'ngInject';
+
+        // Service dependencies region //
+
+        this._storage = $localStorage;
+        this._$filter = $filter;
+        this._Wallet = Wallet;
+
+        // End dependencies region //
+
+        // Service properties region //
+
+        this.apiBaseUrl = NODE_REWARDS_PROGRAM_API_BASE_URL;
+        this.common = nem.model.objects.get('common');
+
+        // End properties region //
+    }
+
+    // Service methods region //
+
+    /**
+     * Prepare the enroll transaction
+     *
+     * @param {string} mainPublicKey - Delegated harvesting public key.
+     * @param {string} host - IP or domain of the node.
+     * @param {string} enrollmentAddress - Node Rewards Program enrollment address.
+     * @param {string} [multisigAccountAddress] - Multisig account address. For multisig enrollment only.
+     *
+     * @return {Promise<object>} - Prepared enroll transaction
+     */
+    async prepareEnrollTransaction(mainPublicKey, host, enrollmentAddress, multisigAccountAddress) {
+        // Create a new transaction object
+        const transferTransaction = nem.model.objects.get('transferTransaction');
+
+        // Set enrollment address as recipient
+        transferTransaction.recipient = enrollmentAddress.toUpperCase().replace(/-/g, '');
+
+        // Set multisig, if selected
+        if (multisigAccountAddress) {
+            transferTransaction.isMultisig = true;
+            transferTransaction.multisigAccount = multisigAccountAddress.toUpperCase().replace(/-/g, '');
+        }        
+
+        // Set the message
+        const hash = await this.getCodewordHash(mainPublicKey);
+        transferTransaction.message = `enroll ${host} ${hash}`;
+
+        // Set no mosaics
+        transferTransaction.mosaics = null;
+
+        return nem.model.transactions.prepare('transferTransaction')(this.common, transferTransaction, this._Wallet.network);
+    }
```

Comment:
I think we can keep this in the controller. 
since this method didn't require `fetch` 

what do you think?

Reply:
> @AnthonyLaw Can you elaborate on this? (Which class are you referring to and why?)

sure, I mean `prepareEnrollTransaction` method, not `class`.

for my understanding, `*.service.js` should just make a request to the endpoint `NODE_REWARDS_PROGRAM_API_BASE_URL`.

but `prepareEnrollTransaction` just prepares the transaction object, it does not have any interaction with another endpoint.

so I think preparing transaction object should happen in `nodeRewardsProgram.controller`.

Reply:
I think this is not the case.
> for my understanding, *.service.js should just make a request to the endpoint NODE_REWARDS_PROGRAM_API_BASE_URL

Please see this answer => https://stackoverflow.com/a/27398868/3190131

And this => https://docs.angularjs.org/guide/services